client/src/route/config/Verify.tsx

```diff
@@ -0,0 +1,229 @@
+import { z } from 'zod';
+import * as React from 'react';
+import { Box, CircularProgress, Paper, Typography } from '@mui/material';
+import { ConfigItem } from 'route/config/ConfigItems';
+
+const EnvironmentEnum = z.enum(['dev', 'local', 'prod', 'test']);
+const PathString = z.string();
+const ScopesString = z.literal('openid profile read_user read_repository api');
+
+const VerifyConfig: React.FC<{ keys?: string[]; title?: string }> = ({
+  keys = [],
+  title = 'Config verification',
+}) => {
+  const [validationResults, setValidationResults] = React.useState<{
+    [key: string]: validationType;
+  }>({});
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  React.useEffect(() => {
+    const fetchValidations = async () => {
+      const results = await getValidationResults(keys);
+      setValidationResults(results);
+      setIsLoading(false);
+    };
+    fetchValidations();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [window.env]);
+
+  const displayedConfigs: Partial<typeof window.env> =
+    keys.length === 0
+      ? window.env
+      : Object.fromEntries(
+          keys
+            .filter((key) => key in window.env)
+            .map((key) => [
+              key,
+              window.env[key as keyof typeof window.env] as string,
+            ]),
+        );
+
+  return isLoading ? (
+    loadingComponent()
+  ) : (
+    <Paper
+      sx={{
+        p: 2,
+        height: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+        position: 'relative',
+      }}
+    >
+      <Typography variant="h4">{title}</Typography>
+      <div id="config-items">
+        {Object.entries(displayedConfigs).map(([key, value]) => (
+          <ConfigItem
+            key={key}
+            label={key}
+            value={value!}
+            validation={validationResults[key]}
+          />
+        ))}
+      </div>
+    </Paper>
+  );
+};
+
+export type validationType = {
```

Comment:
these members are used in `getConfigIcon()` as mandatory parameters. I am surprised that **tsc** has not thrown an error.